.github/workflows/km-ci-workflow.yaml

```diff
@@ -631,3 +670,46 @@ jobs:
           channel: "#build_and_test"
           name: "CI workflow status"
           icon_emoji: ":thumbsdown:"
+
+
+  # These two steps are to clean up on-demand runner. They work in conjunction with start-runner.
+  # Make sure to add dependencies in both "needs" clauses
+  stop-runner:
```

Comment:
`stop-runner` gets invoked when the steps in it's `needs:` are finished. `[start-runner, km-build, kkm-test-vms]`

Reply:
Jobs are simply executed by github, all of them. They will run in parallel unless there are dependencies. For instance this one needs `start-runner`, `km-build` and `kkm-test-vms` so it will run after all of them complete.
`if: ${{ ! failure() }}` clause makes this run only on success or cancel. `stop-and-keep-runner` job is complementary, it will run on failures. I mean they both get started by github but only one proceeds to do anything because of `if:` clauses